### ISSUE TITLE
Support for web interfaces of various services

### ISSUE THREAD

**Issue by allisonkarlitskaya** [cockpit-project/cockpit#18674]
Pie in the sky feature idea:

I use `cockpit-podman` to run a number of services as containers on a small home server:
 - home-assistant
 - jellyfin
 - unifi controller

One thing I start to notice is that all of these things have web interfaces of their own and doing "admin stuff" on my server often involves interacting with those interfaces.

Home Assistant OS supports add-ons in the form of containers and part of the specification of a add-on involves being able to declare a web "ingress".  That's documented here: https://developers.home-assistant.io/docs/add-ons/presentation/

There's also a nice video here: https://www.home-assistant.io/blog/2019/04/15/hassio-ingress/

You get a nice "Open Web UI" button from the container config.  I think this works even if the web port of the container isn't publicly available.  It can also be done in a way that bypasses authentication, if the service supports that.  Newer versions of Home Assistant can also add the web interface for a given add-on to the left navigation panel.

It would be pretty interesting to have some feature like this in Cockpit as a way to present access to running containers.  We might imagine some kind of a label on the container as metadata for expressing the availability of a web interface on a given port.

**Comment by garrett** [comment]
Here's a quick mockup of a modal at how adding an app could work:

![add application mockup](https://user-images.githubusercontent.com/10246/233410592-17233ee8-2212-4806-80e0-d3f23e4d64fa.png)

We'd want to add functionality for this to the applications page. And we'd want to make a different version with location prefilled and selectable for Cockpit-Podman, which would help automate this process and add it directly from the container row. (Possibly the same for Cockpit-Machines too, really.)

**Comment by champtar** [comment]
If it's ever implemented, please add an option for dynamic hostname (`window.location.hostname`), and possibility to add it dynamically via configuration in /run somewhere.

Right now I use cockpit to enable / disable some 'products' to run on the server + dynamically update a manifest.json + template some html + call `parent.location.reload(true)`, it works but is hacky as hell

**Comment by michacassola** [comment]
Was there any planning regarding this since last year @garrett ?

**Comment by garrett** [comment]
@michacassola: No, there has been no planning on this whatsoever. We're a really small team at the moment, and have a bunch of other priorities.

If anyone wants to step up and work on this, we'd like to help out.
